server/src/main/java/io/crate/metadata/upgrade/MetadataIndexUpgrader.java

```diff
@@ -90,4 +113,76 @@ MappingMetadata createUpdatedIndexMetadata(MappingMetadata mappingMetadata, Stri
             return mappingMetadata;
         }
     }
+
+    private void reorderPositionsIfNecessary(Map<String, Object> properties) {
+        if (shouldReorder(properties)) {
+            reorderPositions(properties, 1);
+        }
```

Comment:
I think instead of mutating the properties this should return a new map if there was a change. I'm not sure if it's always safe to assume that the map can be mutated.
It could also avoid the mapping update if there is none, instead of building a new indexMetaata even if there was no change.